string                              id
uint32                              bid
uint32                              expected_steps
string                              agent_name
string[]               items